### PR TITLE
MES-2948: fix pass cert length

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description and relevant Jira numbers
- update `pass-finalisation` validation to enforce a max length of 8 chars for `pass-certificate-numbers`

Screenshot 2019-07-19 at 12 09 32
Screenshot 2019-07-19 at 12 09 43

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/pass-finalisation/__tests__/pass-finalisation.spec.ts [hunk 1]
```diff
@@ -59,10 +59,34 @@ describe('PassFinalisationPage', () => {
...
       const form = component.form;
       form.get('provisionalLicenseProvidedCtrl').setValue(true);
-      form.get('passCertificateNumberCtrl').setValue(true);
+      form.get('passCertificateNumberCtrl').setValue('A123456*');
       form.get('transmissionCtrl').setValue('Manual');
       component.onSubmit();
       expect(store$.dispatch).toHaveBeenCalledWith(new PersistTests());
     });
   });
+  fdescribe('formControls', () => {
```
Comment (johnmorleybjss) [inline src/pages/pass-finalisation/__tests__/pass-finalisation.spec.ts]: Focused test suite needs removing.

Comment (MarkPatt89) [reply src/pages/pass-finalisation/__tests__/pass-finalisation.spec.ts]: ahh, thanks John.

File: src/pages/pass-finalisation/pass-finalisation.ts [hunk 2]
```diff
@@ -187,16 +187,26 @@ export class PassFinalisationPage extends PracticeableBasePageComponent {
   }
 
   getFormValidation(): { [key: string]: FormControl } {
+    const passCerticateFormat = new RegExp(/\D{1}\d{6}\W{1}/);
```
Comment (johnmorleybjss) [inline src/pages/pass-finalisation/pass-finalisation.ts]: Won't this accept symbols for the first character (anything that's not a digit)? Shouldn't you use `[A-Z]`?

File: src/pages/pass-finalisation/__tests__/pass-finalisation.spec.ts (hunk 1 above)
Comment (MarkPatt89) [reply src/pages/pass-finalisation/__tests__/pass-finalisation.spec.ts]: Done

File: src/pages/pass-finalisation/pass-finalisation.ts (hunk 2 above)
Comment (MarkPatt89) [reply src/pages/pass-finalisation/pass-finalisation.ts]: You're 100% right, fixed and pushing changes once I check out the comment above.

Comment (MarkPatt89) [reply src/pages/pass-finalisation/pass-finalisation.ts]: Spoke with Andy around the last digit, this has also been amended.

**Comment by buzzub** [comment]
Shouldn't the error message say what the allowed / expected format is? E.g. https://design-system.service.gov.uk/components/error-summary/ shows "Enter a postcode, like AA1 1AA"

**Comment by MarkPatt89** [comment]
> Shouldn't the error message say what the allowed / expected format is? E.g. https://design-system.service.gov.uk/components/error-summary/ shows "Enter a postcode, like AA1 1AA"

I had queried this, they will have the number on the certificate in front of them. Also verified the content with Oli.

**Comment by buzzub** [comment]
I know that the examiner will have the certificate in front of them (and this error is unlikely to appear), but I feel that we should let them know if they should be prefixing with the letter or not, especially if we're asking for a "**pass certificate number**".

I think we either need to include an example in the field label or an explanation in the error text. As it stands, the error does not help the user resolve it.

**Comment by MarkPatt89** [comment]
> I know that the examiner will have the certificate in front of them (and this error is unlikely to appear), but I feel that we should let them know if they should be prefixing with the letter or not, especially if we're asking for a "**pass certificate number**".
> 
> I think we either need to include an example in the field label or an explanation in the error text. As it stands, the error does not help the user resolve it.

I'm happy to change it, I'd rather the design guys approval first. I'll query it with them.

**Comment by buzzub** [comment]
> I'm happy to change it, I'd rather the design guys approval first. I'll query it with them.

Just cite this from the GDS guide: 

**> Above all, aim for clarity.**

**Comment by MarkPatt89** [comment]
> Looks good Mark - do we need to do anything to enforce this on the test result service or via the schema

No, we don't need to concern ourselves with that. Catching this at the application layer will suffice. The issue was spotted by Chris, as RSIS MI DB was rejecting test results due to them not meeting the desired pattern.

**Pull request merged**